packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.tsx

```diff
@@ -736,6 +735,20 @@ export function TransferPanel() {
     disableWithdrawal
   ])
 
+  // a common set-amount function to set both the deposit and withdrawal amounts
+  // helpful when we want to set both l1, l2 amounts when browser query params change
+  const setAmount = useCallback(amount => {
+    setl1Amount(amount || '0')
+    setl2Amount(amount || '0')
+  }, [])
```

Comment:
Can we ditch `useState` completely for amounts and just rely on what's in query params? 🤔 
This way we keep state in 2 places and we have to be careful to sync them.

Reply:
- Both `l1Amount` and `l2Amount` (the state values) are being used extensively in the `TransferPanel` code.

- These are the state variables for the input element present in the transfer panel and I don't think we can bypass them to directly link to the query-params. Maybe I didn't understand your pov clearly, we can connect.

- However, I see a lot of redundancy in code trying to maintain `l1Amount` and `l2Amount`, separately (and for common use-cases, merging them into a separate `amount` state) and I think we can do well with only maintaining the `amount state (amount, setAmount)` only.